cpp/Libusb.cpp

```diff
@@ -72,6 +72,19 @@ std::variant<libusb::Claimed_interface, libusb::Error> libusb::Open_device::clai
     return libusb::Claimed_interface(inner, interface_number);
 }
 
+std::variant<int, libusb::Error> libusb::Open_device::wait_for_config() {
+    int r = -1;
+    int config = -1;
+    const int max_retries = 5;
+    libusb_device_handle *devh2 = inner->devh;
+
+    for (int i = 0; i < (i < max_retries && ((r != 0 || (r == 0 && (config == 0 || config == -1))))); i++) {
```

Comment:
This could be more readable.

Reply:
What about simplifying:
(r != 0 || (r == 0 && (config == 0 || config == -1)))
(r != 0 || config <= 0)

Reply:
Is this correct?
i < (i < max_retries.... ?
Should the outer "i < " be deleted?